packages/fluentui/react-northstar/src/themes/teams/components/Button/buttonStyles.ts

```diff
@@ -207,11 +207,13 @@ const buttonStyles: ComponentSlotStylesPrepared<ButtonStylesProps, ButtonVariabl
         minWidth: v.height,
         padding: 0,
 
-        ':hover': {
-          ...getIconFillOrOutlineStyles({ outline: false }),
-          color: v.textColorIconOnlyHover,
-          background: v.backgroundColorIconOnlyHover,
-        },
+        ...(!p.inverted && {
+          ':hover': {
+            ...getIconFillOrOutlineStyles({ outline: false }),
+            color: v.textColorIconOnlyHover,
+            background: v.backgroundColorIconOnlyHover,
+          },
+        }),
```

Comment:
```suggestion
        ':hover': {
          ...getIconFillOrOutlineStyles({ outline: false }),
          color: v.textColorIconOnlyHover,
          background: v.backgroundColorIconOnlyHover,
        },
        ...(p.inverted && { ':hover': undefined })
```

Let's avoid negative conditions in styles.

Reply:
Actually, the change doesn't work because when `p.inverted` is true, the color and background on `:hover` should be set to:

```js
{
  color: siteVariables.colorScheme.silver.foregroundHover,
  backgroundColor: siteVariables.colorScheme.silver.backgroundHover,
}
```

which is what is was in the [first commit](https://github.com/microsoft/fluentui/pull/13198/commits/70b16921cf6dcf88aa579f29bc230fe9f6eb0762) of the PR which also did not have negative conditions, but instead reinforced the inverted background & foreground colors as @notandrew said